numpy/_core/strings.py

```diff
@@ -1638,3 +1639,89 @@ def translate(a, table, deletechars=None):
+    """
+    Slice the strings in `a` by slices specified by `start`, `stop`, `step`.
+    Like in the regular Python `slice` object, if only `start` is
+    specified then it is interpreted as the `stop`.
+
+    Parameters
+    ----------
+    a : array-like, with ``StringDType``, ``bytes_``, or ``str_`` dtype
+
+    start : the start of the slice, can be None, an integer or
+            an array of integers which can be broadcasted to`a`'s shape
+
+    stop : the end point of the slice, can be None, and integer or
+           an array of integers which can be broadcasted to`a`'s shape
+
+    step : the step for the slice, can be None, an integer or
+           an array of integers which can be broadcasted to`a`'s shape
+
+    Returns
+    -------
+    out : ndarray
+        Output array of str or unicode, depending on input type
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> a = np.array(['hello', 'world'])
+    >>> np.char.slice(a, 2)
+    array(['he', 'wo'], dtype='<U5')
+
+    >>> np.char.slice(a, 1, 5, 2)
+    array(['el', 'ol'], dtype='<U5')
+
+    One can specify different start/stop/step for different array entries:
+
+    >>> np.strings.slice(a, np.array([1, 2]), np.array([4, 5]))
+    array(['ell', 'rld'], dtype='<U5')
+
+    Negative slices have the same meaning as in regular Python:
+
+    >>> b = np.array(['hello world', 'γεια σου κόσμε', '你好世界', '👋 🌍'],
+    ...              dtype=np.dtypes.StringDType())
+    >>> np.strings.slice(b, -2)
+    array(['hello wor', 'γεια σου κόσ', '你好', '👋'], dtype=StringDType())
+
+    >>> np.strings.slice(b, [3, -10, 2, -3], [-1, -2, -1, 3])
+    array(['lo worl', ' σου κόσ', '世', '👋 🌍'], dtype=StringDType())
+
+    >>> np.strings.slice(b, None, None, -1)
+    array(['dlrow olleh', 'εμσόκ υοσ αιεγ', '界世好你', '🌍 👋'],
+          dtype=StringDType())
+
+    """
+    # Just like in the construction of a regular slice object, if only start
+    # is speficied then start will become stop, see logic in slice_new.
+    if stop is None:
+        stop = start
+        start = None
+
+    # adjust start, stop, step to be integers, see logic in PySlice_Unpack
+    if step is None:
+        step = 1
+    step = np.asanyarray(step)
+    if not np.issubdtype(step.dtype, np.integer):
+        raise TypeError(f"unsupported type {step.dtype} for operand 'step'")
+    if np.any(step == 0):
+        raise ValueError("slice step cannot be zero")
+
+    if start is None:
+        start = np.where(step < 0, np.iinfo(np.intp).max, 0)
+    start = np.asanyarray(start)
+    if not np.issubdtype(start.dtype, np.integer):
+        raise TypeError(f"unsupported type {start.dtype} for operand 'start'")
+
+    if stop is None:
+        stop = np.where(step < 0, np.iinfo(np.intp).min, np.iinfo(np.intp).max)
+    stop = np.asanyarray(stop)
+    if not np.issubdtype(stop.dtype, np.integer):
+        raise TypeError(f"unsupported type {stop.dtype} for operand 'stop'")
```

Comment:
```suggestion
    if start is None:
        start = np.where(step < 0, np.iinfo(np.intp).max, 0)
    else:
        start = np.asanyarray(start)
        if not np.issubdtype(start.dtype, np.integer):
            raise TypeError(f"unsupported type {start.dtype} for operand 'start'")

    if stop is None:
        stop = np.where(step < 0, np.iinfo(np.intp).min, np.iinfo(np.intp).max)
    else:
        stop = np.asanyarray(stop)
        if not np.issubdtype(stop.dtype, np.integer):
            raise TypeError(f"unsupported type {stop.dtype} for operand 'stop'")
```

no need to call `asanyarray` or check dtypes if you create an array from `np.where`, I think.